src/main/java/gregtech/common/covers/CoverFluidRegulator.java

```diff
@@ -0,0 +1,245 @@
+package gregtech.common.covers;
+
+import gregtech.api.cover.ICoverable;
+import gregtech.api.gui.GuiTextures;
+import gregtech.api.gui.ModularUI;
+import gregtech.api.gui.widgets.*;
+import gregtech.api.util.GTUtility;
+import net.minecraft.client.entity.EntityPlayerSP;
+import net.minecraft.entity.player.EntityPlayer;
+import net.minecraft.nbt.NBTTagCompound;
+import net.minecraft.tileentity.TileEntity;
+import net.minecraft.util.EnumFacing;
+import net.minecraft.util.math.BlockPos;
+import net.minecraft.util.math.MathHelper;
+import net.minecraft.util.text.ITextComponent;
+import net.minecraft.util.text.TextComponentString;
+import net.minecraft.util.text.TextComponentTranslation;
+import net.minecraft.util.text.event.HoverEvent;
+import net.minecraftforge.fluids.FluidStack;
+import net.minecraftforge.fluids.capability.CapabilityFluidHandler;
+import net.minecraftforge.fluids.capability.IFluidHandler;
+import net.minecraftforge.fluids.capability.IFluidTankProperties;
+
+import java.util.List;
+import java.util.function.Predicate;
+
+import static gregtech.api.gui.widgets.AdvancedTextWidget.withHoverTextTranslate;
+
+public class CoverFluidRegulator extends CoverPump {
+
+    protected TransferMode transferMode;
+    protected int keepAmount = 0;
+    protected int supplyAmount = 0;
+    private static final String supplyKey = "SupplyAmount";
+    private static final String keepKey = "KeepAmount";
+
+    public CoverFluidRegulator(ICoverable coverHolder, EnumFacing attachedSide, int tier, int mbPerTick) {
+        super(coverHolder, attachedSide, tier, mbPerTick);
+        this.transferMode = TransferMode.TRANSFER_ANY;
+    }
+
+    @Override
+    protected int doTransferFluids(int transferLimit) {
+        BlockPos.PooledMutableBlockPos blockPos = BlockPos.PooledMutableBlockPos.retain();
+        blockPos.setPos(coverHolder.getPos()).move(attachedSide);
+        TileEntity tileEntity = coverHolder.getWorld().getTileEntity(blockPos);
+        blockPos.release();
+        IFluidHandler fluidHandler = tileEntity == null ? null : tileEntity.getCapability(CapabilityFluidHandler.FLUID_HANDLER_CAPABILITY, attachedSide.getOpposite());
+        IFluidHandler myFluidHandler = coverHolder.getCapability(CapabilityFluidHandler.FLUID_HANDLER_CAPABILITY, attachedSide);
+        if (fluidHandler == null || myFluidHandler == null) {
+            return 0;
+        }
+        IFluidHandler sourceHandler;
+        IFluidHandler destHandler;
```

Comment:
This is already handled by `doTransferFluids` on Pump you should just override `doTransferFluidsInternal` and continue form there.